Assets/Scripts/Data/Pickup Pup/Currency/HomeSlotsData.cs

```diff
@@ -0,0 +1,157 @@
+﻿/*
+ * Author: Grace Barrett-Snyder 
+ * Description: Stores data for the Player's home slots currency.
+ */
+
+[System.Serializable]
+public class HomeSlotsData : CurrencyData
+{
+    HomeSlot[] slots;
+
+    #region Instance Accessors
+
+    public int VacantSlots
+    {
+        get
+        {
+            return countSlotsWithStatus(HomeSlotStatus.Vacant);
+        }
+    }
+
+    public bool HasVacantSlot
+    {
+        get
+        {
+            return hasSlotWithStatus(HomeSlotStatus.Vacant);
+        }
+    }
+
+    public int OccupiedSlots
+    {
+        get
+        {
+            return countSlotsWithStatus(HomeSlotStatus.Occupied);
+        }
+    }
+
+    public bool HasOccupiedSlot
+    {
+        get
+        {
+            return hasSlotWithStatus(HomeSlotStatus.Occupied);
+        }
+    }
+
+    #endregion
+
+    public HomeSlotsData(int initialAmount) : base(initialAmount)
+    {
+        type = CurrencyType.HomeSlots;
+        amount = initialAmount;
+        slots = new HomeSlot[amount];
+        for (int i = 0; i < slots.Length; i++)
+        {
+            slots[i] = new HomeSlot(HomeSlotStatus.Vacant);
+        }
+    }
+
+    #region CurrencyData Overrides
+
+    // Only the vacant slots are exchanged as currency
+    // Ex: you need a vacant slot to adopt a dog
+    public override bool CanAfford(int cost)
```

Comment:
What if in future a dog costs more than one vacant slot? Should likely still be an `int` check